app.rb

```diff
@@ -0,0 +1,111 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'rentals'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_all_books(_books)
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+      puts ' '
+    end
+  end
+
+  def list_all_people(_people)
+    @people.each do |person|
+      print "[#{person.class}] name: #{person.name}, ID: #{person.id}, #{person.age}"
+      puts ' '
+    end
+  end
+
+  def create_student
+    print 'Age:'
+    age = gets.chomp
+    print 'name:'
+    name = gets.chomp
+    print 'Has parent permission? [Y/N]:'
+    permit = gets.chomp
+    case permit
+    when 'Y', 'y'
+      permission = true
+    when 'N', 'n'
+      permission = false
+    end
+    student = Student.new(age, name, parent_permission: permission)
+    @people.push(student)
+    puts 'Person successfully created'
+  end
+
+  def create_teacher
+    print 'Age:'
+    age = gets.chomp
+    print 'name:'
+    name = gets.chomp
+    print 'specialization:'
+    specialization = gets.chomp
+    teacher = Teacher.new(specialization, age, name)
+    @people.push(teacher)
+    puts 'Person created successfully'
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher(2)? [Input the number]'
+    response = gets.chomp
+    case response
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    end
+  end
+
+  def create_book
+    print 'Title:'
+    title = gets.chomp
+    print 'Author:'
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'book created successfully'
+  end
```

Comment:
- [ ] At testing your app I couldn't create a book, please review your code to avoid this error 👇🏽 

![image](https://user-images.githubusercontent.com/58642949/218267446-ae5e1413-4524-4281-8892-d22f7c32af67.png)